src/api/modules/dispatch/dispatch.background.ts

```diff
@@ -153,6 +157,10 @@ const background: BackgroundModuleFunction<ReturnType> = async (
         type: "BASE",
       },
     };
+  } catch (err) {
+    freeDecryptedWallet(keyfile);
```

Comment:
With the previous implementation with a single `try-catch`, if the first upload attempt fails, the second one would fail too, because `ensureAllowanceDispatch()` internally had another `try-catch` that would call `freeDecryptedWallet()`, so in the second upload attempt there would be no keyfile.

If we just update `ensureAllowanceDispatch()` to get rid of that internal `try-catch` and call to `freeDecryptedWallet()`, then this single `try-catch` here becomes insecure because if the code inside the `catch` block fails, `freeDecryptedWallet()` might never be called, thus, we need to use 2 separate `try-catch` blocks here.